### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by sberyozkin**

### CODE REVIEW

File: implementation/pom.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  <artifactId>smallrye-health</artifactId>
+
+  <name>SmallRye: MicroProfile Health Implementation</name>
+
+  <dependencies>
+    <dependency>
+      <groupId>org.eclipse.microprofile.health</groupId>
+      <artifactId>microprofile-health-api</artifactId>
+    </dependency>
+    <dependency>
```
Comment (kenfinnigan) [inline implementation/pom.xml]: Should servlet, cdi and json dependencies be provided scope?

**Comment by jmesnil** [comment]
This implementation is based on Servlet API that is not a specification provided for MicroProfile.

In my case, I was planning to use Undertow handler to provide the HTTP endpoint.
I wonder if this implementation should provide a SPI so that integrators can plug it into their web stack (whether it is based on Undertow, Servlet, or other lib/frameworks)

**Comment by kenfinnigan** [comment]
For the initial version I'm ok with it being tightly coupled, but agree we should provide something more clever going forward.

@jmesnil, can you create an issue in the repo describing your proposal?

File: implementation/pom.xml (hunk 1 above)
Comment (sberyozkin) [reply implementation/pom.xml]: I haven't thought about it, I can try to see how it works against v2 and v4

**Comment by sberyozkin** [comment]
Yes, I thought that was something that would be easier to look at it once the external integrators will look at reusing it. I thought that 'servlet' may not be the common implementation idea, possibly JAX-RS instead.

@jmesnil, thanks, though I did not understand your comment about Servlet impl being not the Spec provided, the spec does not talk about the impl details ?

**Comment by jmesnil** [comment]
@sberyozkin sorry, I was not very clear.

The microprofile umbrella specification does not include the servlet API. I don't think that this implementation of the health spec should mandate to use it.
In my case (WildFly) we are interested to implement the health check API in our low-level kernel code using Undertow handler. We will not have access to the servlet API there.

I'll open an issue with some proposal about this point.

**Comment by sberyozkin** [comment]
@jmesnil thanks, yes, this implementation is just once concrete instance of the Health API implementation based on, in this case, Servlet API. Right, so you'd like to reuse only the implementation part related to managing the HealthCheck calls, etc...Sounds good

**Comment by sberyozkin** [comment]
Though I guess some default implementation should be there anyway. I guess it can be discussed later :-)

File: implementation/pom.xml (hunk 1 above)
Comment (sberyozkin) [reply implementation/pom.xml]: Ken, looks like adding the provided scope worked well, v2 and v4 tcks still pass on my branches

**Comment by sberyozkin** [comment]
@jmesnil, hi, please see #2, this should me it simpler to integrate; more improvements can be done but I hope it is a good start :-)

**Pull request merged**